### PR TITLE
Drag and drop image to the file browser through the canvas

### PR DESCRIPTION
**Pull request by bkrmendy**
## Problem:
See detailed description in https://github.com/concrete-utopia/utopia/issues/2650

## Fix:
Disable the cursor overlay when a drag session is in progress, so that it doesn't capture mouse events. This can be done by creating a new state on EditorState (`dragSessionState`) and the associated action (`SET_DRAG_SESSION_STATE`). This state is maintained when starting/ending drag sessions, and it can be checked by the cursor overlay.

**Commit Details:**
- Modify the cursor overlay so that it doesn't capture mouse events
- Create a new state on EditorState (`dragSessionState`) and update the boilerplate
- Create a new action (`SET_DRAG_SESSION_STATE`) with the associated boilerplate
- read `dragSessionState` in `CursorOverlay` to catch/pass pointer events
- set ``dragSessionState` in `FileItem` and `EditorCanvas` during drag sessions
- update tests to work with the new setup

### CODE REVIEW

File: editor/src/templates/editor-canvas.tsx [hunk 1]
```diff
@@ -943,8 +963,18 @@ export class EditorCanvas extends React.Component<EditorCanvasProps> {
...
+              EditorActions.setFileBrowserDragState(null),
+              CanvasActions.clearInteractionSession(true),
+              EditorActions.switchEditorMode(EditorModes.selectMode()),
+            ])
+            return
+          }
           event.preventDefault()
           event.stopPropagation()
+
+          // itt kene tudni a filebrowserben melyik kepet draggeljuk, es akkor nem kell feltolteni
```
Comment (bkrmendy) [inline editor/src/templates/editor-canvas.tsx:977]: remove

File: editor/src/components/editor/store/editor-state.ts [hunk 2]
```diff
@@ -752,6 +752,54 @@ export function editorStateInspector(
...
+interface DraggingFromFS {
+  type: 'DRAGGING_FROM_FS'
+}
+
+export interface DraggingFromSidebar {
+  type: 'DRAGGING_FROM_SIDEBAR'
+  draggedImageProperties: DraggedImageProperties
+}
+
+export type DragSessionState = NotDragging | DraggingFromFS | DraggingFromSidebar
```
Comment (Rheeseyb) [inline editor/src/components/editor/store/editor-state.ts:774]: Please can we rename this to reflect the fact it is specifically for asset dragging (or just image dragging?)

File: editor/src/components/editor/action-types.ts [hunk 3]
```diff
@@ -998,6 +1000,11 @@ export interface SaveToGithub {
   targetRepository: GithubRepo
 }
 
+export interface SetFileBrowserDragState {
+  action: 'SET_DRAG_SESSION_STATE'
```
Comment (Rheeseyb) [inline editor/src/components/editor/action-types.ts:1004]: Please can we also change this to reflect that it is specific to the file browser / image / asset dragging

File: editor/src/templates/editor-canvas.tsx [hunk 4]
```diff
@@ -909,42 +915,70 @@ export class EditorCanvas extends React.Component<EditorCanvasProps> {
...
-                createInteractionViaMouse(
-                  position.canvasPositionRounded,
-                  emptyModifiers,
-                  boundingArea(),
-                ),
-              ),
-            ],
-            'everyone',
+          const newElementProps: Partial<DraggedImageProperties> =
+            this.props.editor.dragSessionState.type === 'DRAGGING_FROM_SIDEBAR'
```
Comment (Rheeseyb) [inline editor/src/templates/editor-canvas.tsx:928]: It looks like this condition should probably be used for an early return from the `onDragOver` event

Comment (Rheeseyb) [reply editor/src/templates/editor-canvas.tsx:928]: That would also mean the above `Partial<DraggedImageProperties>` can become `DraggedImageProperties`, meaning no need for the null fallbacks below

File: editor/src/templates/image-dnd.spec.browser2.tsx [hunk 5]
```diff
@@ -240,7 +241,7 @@ describe('image dnd', () => {
     sandbox.restore()
   })
 
-  it('dragging from the sidebar works', async () => {
+  xit('dragging from the sidebar works', async () => {
```
Comment (Rheeseyb) [inline editor/src/templates/image-dnd.spec.browser2.tsx:244]: Should this test still be disabled?

Comment (bkrmendy) [reply editor/src/templates/image-dnd.spec.browser2.tsx:244]: it's disabled on as long as https://github.com/concrete-utopia/utopia/pull/2659 isn't merged, so once it is, I'm aiming to fix it in a followup PR

File: editor/src/components/editor/store/editor-state.ts (hunk 2 above)
Comment (bkrmendy) [reply editor/src/components/editor/store/editor-state.ts]: absolutely, renamed it

File: editor/src/components/editor/action-types.ts (hunk 3 above)
Comment (bkrmendy) [reply editor/src/components/editor/action-types.ts]: renamed it

**Review by seanparsons — APPROVED** [review]

File: editor/src/templates/editor-canvas.tsx (hunk 4 above)
Comment (bkrmendy) [reply editor/src/templates/editor-canvas.tsx]: discussed on DC: https://discord.com/channels/647102474519576589/1024328724268732476/1030136332971946027

**Review by Rheeseyb — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

**Pull request merged**